### ISSUE TITLE
We could not retrieve data from the specified Element.

### ISSUE THREAD

**Issue by UndercoverNL** [stripe/react-stripe-js#334]
### What happened?

So I am having a complicated issue running a CardElement with a loader.
As there is no indication of the Element form being submitted by default, I am trying to use a simple loader `useState` which will indicate the form being submitted and ensure it won't be submitted multiple times by the user.

However, as `setLoading(true)` causes a re-render it errors the following in the console:
![image](https://user-images.githubusercontent.com/42718013/196012750-f8807c39-3007-4980-89ce-879eed22ab0c.png)

This is probably due to stripe being loaded multiple times. As when I remove the `setLoading(true)` all works, but without loader.

```
const submit = async (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    clearFlashes('account:payment-methods:create');
    setLoading(true);

    if (elements == null || stripe == null) {
        return;
    }

    const { error, paymentMethod } = await stripe.createPaymentMethod({
        type: 'card',
        card: elements.getElement(CardElement),
    });

    if (error) return clearAndAddHttpError({ key: 'account:payment-methods:create', error });

....
};
```

this is my current code. The setLoading state is declared as: `const [ loading, setLoading ] = useState(false);`

Is there any way to set an useState or any other kind as loader for the CardElement form?

### Environment

Version 1.44.112 Chromium: 106.0.5249.119 (Official Build) (64-bit) Windows 10

**Comment by bmathews-stripe** [comment]
If you're already doing something like the latter example, make sure the element isn't unmounted for some other reason, like a changing `key` on an element or component higher up, or functional component defined in the render method of another functional component.

The library is likely working as intended, though. `createPaymentMethod` requires the element to still be mounted in order to pull data from it.

If you can create a small reproduction example on something like codesandbox, we can re-open this and go from there.

**Comment by UndercoverNL** [comment]
@bmathews-stripe the key of the CardElement could also be a string? Instead of a variable string? As even when I use a string as key it has the same behaviour.
And I am already doing it the way you send it.

The thing causing the error is the `setLoading(true)` inside the submit event.
Of which I thought it was due to the setLoading triggering a re-render, making the module load multiple which times, which was one of the causes of this error according to some research I did. However, if this really was the case the package would be very React unfriendly if it won't work with hooks and re-renders, as that is what react uses all over the place.
